internal/literals/literals.go

```diff
@@ -130,37 +130,42 @@ func Obfuscate(file *ast.File, info *types.Info, fset *token.FileSet, linkString
 		return true
 	}
 
-	// Imports which are used might be marked as unused by only looking at the ast,
-	// because packages can declare a name different from the last element of their import path.
-	//
-	// 	package main
-	//
-	// 	import "github.com/user/somepackage"
-	//
-	//	func main(){
-	//		// this line uses github.com/user/somepackage
-	//		anotherpackage.Foo()
-	//	}
-	//
-	// TODO: remove this check and detect used imports with go/types somehow
-	prevUsedImports := make(map[string]bool)
-	for _, imp := range file.Imports {
-		path, err := strconv.Unquote(imp.Path.Value)
-		if err != nil {
-			panic(err)
-		}
-		prevUsedImports[path] = astutil.UsesImport(file, path)
-	}
-
 	file = astutil.Apply(file, pre, post).(*ast.File)
 
 	// some imported constants might not be needed anymore, remove unnecessary imports
+	usedImports := make(map[string]bool)
+	ast.Inspect(file, func(n ast.Node) bool {
+		node, ok := n.(*ast.SelectorExpr)
+		if !ok {
+			return true
+		}
+
+		x, ok := node.X.(*ast.Ident)
+		if !ok {
+			return true
+		}
+
+		uses, ok := info.Uses[x].(*types.PkgName)
+		if !ok {
+			return true
+		}
+
+		usedImports[uses.Imported().Path()] = true
+
+		return true
+	})
+
 	for _, imp := range file.Imports {
+		if imp.Name != nil && (imp.Name.Name == "_" || imp.Name.Name == ".") {
```

Comment:
Hmm, I think we need a TODO for dot imports, because they could become unused just like any other import. Luckily, I expect almost noone to use dot imports, so we can leave that as a semi-permanent TODO.